tests/integration/test_aws/test_regions.py

```diff
@@ -141,35 +139,34 @@ def test_regions(
             callback=event_monitor.callback_detect_event_processed,
             accumulations=expected_results
         )
-
         assert log_monitor.callback_result is not None, ERROR_MESSAGE['incorrect_event_number']
+        
+        assert path_exist(path=S3_CLOUDTRAIL_DB_PATH)
 
-    else:
-        log_monitor.start(
-            timeout=TIMEOUT[10],
-            callback=event_monitor.make_aws_callback(pattern),
-        )
-
-        assert log_monitor.callback_result is not None, ERROR_MESSAGE['incorrect_no_region_found_message']
-
-    assert path_exist(path=S3_CLOUDTRAIL_DB_PATH)
-
-    if expected_results:
+        # Validate database updates
         regions_list = regions.split(",")
         for row in get_multiple_s3_db_row(table_name=bucket_type):
             if hasattr(row, "aws_region"):
                 assert row.aws_region in regions_list
             else:
                 assert row.log_key.split("/")[3] in regions_list
+
     else:
+        log_monitor.start(
+            timeout=session_parameters.default_timeout,
+            callback=event_monitor.make_aws_callback(
+                fr".*\+\+\+ ERROR: Invalid region '{regions}'"
```

Comment:
If the `regions` variable contains more than one region this will fail, because the log will contain the invalid region and this is checking for all of them (`regions`).

For example: 

```
regions = 'us-east-1,us-fake-1'
actual_log = 'Invalid region us-fake-1'
expected_log = 'Invalid region us-east-1,us-fake-1'
```

We should find a way of expecting the first invalid region in the log message only.

Reply:
The test was modified to use only the name of the fake region